test/unit/test_devices/test_ios_device.py

```diff
@@ -1172,3 +1174,61 @@ def test_boot_options_show_boot(mock_boot, show_boot_out, ios_send_command):
     boot_options = device.boot_options
     assert boot_options == {"sys": BOOT_IMAGE}
     device.native.send_command.assert_called_with(command_string="show boot")
+
+
+def test_connected_with_fast_cli():
+    with mock.patch.object(IOSDevice, "confirm_is_active") as mock_confirm:
+        mock_confirm.return_value = True
+        with mock.patch("pyntc.devices.ios_device.ConnectHandler") as ch:
+            device = IOSDevice("host", "user", "password")
+            device.native = ch
+    assert device.fast_cli
+
+
+def test_connected_with_fast_cli_false():
+    with mock.patch.object(IOSDevice, "confirm_is_active") as mock_confirm:
+        mock_confirm.return_value = True
+        with mock.patch("pyntc.devices.ios_device.ConnectHandler") as ch:
+            device = IOSDevice("host", "user", "password", fast_cli=False)
+            device.native = ch
+    assert not device.fast_cli
+
+
+@pytest.mark.parametrize("expected", ((True,), (False,)))
+def test_fast_cli(expected, ios_device):
+    ios_device._fast_cli = expected
+    assert ios_device.fast_cli is expected
+
+
+@pytest.mark.parametrize("expected", ((True,), (False,)))
+def test_fast_cli_setter(expected, ios_device):
+    ios_device._fast_cli = not expected
+    assert ios_device._fast_cli is not expected
+    ios_device.fast_cli = expected
+    assert ios_device._fast_cli is expected
+
+
+def test_image_booted_bundle_version(ios_show):
+    device = ios_show(["show_version.txt"])
+    assert device._image_booted(image_name="c3750-ipservicesk9-mz.150-2.SE11.bin")
+
+
+def test_image_booted_bundle_version_false(ios_show):
+    device = ios_show(["show_version.txt"])
+    assert not device._image_booted(image_name="c3750-ipservicesk9-mz.150-2.SE12.bin")
+
+
+@mock.patch.object(IOSDevice, "show")
+def test_image_booted_install_mode(mock_show, ios_device):
+    with open("test/unit/test_devices/device_mocks/ios/show_version_install_mode.txt") as fh:
+        mock_show.side_effect = [fh.read()]
+
+    assert ios_device._image_booted(image_name="c3750-ipservicesk9-mz.16.09.03.SPA.bin")
+
+
+@mock.patch.object(IOSDevice, "show")
+def test_image_booted_install_mode_fail(mock_show, ios_device):
+    with open("test/unit/test_devices/device_mocks/ios/show_version_install_mode.txt") as fh:
+        mock_show.side_effect = [fh.read()]
+
+    assert not ios_device._image_booted(image_name="c3750-ipservicesk9-mz.16.09.04.SPA.bin")
```

Comment:
The tests are good. However, since you made changes to `file_copy` I suggest that you test these changes.